test/c/test_optimize_1q_decomposition.c

```diff
@@ -0,0 +1,315 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+#include "common.h"
+#include <complex.h>
+#include <math.h>
+#include <qiskit.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+// Helper headers
+QkTarget *get_u1_u2_u3_target(void);
+QkTarget *get_rz_rx_target(void);
+QkTarget *get_rz_sx_target(void);
+QkTarget *get_rz_ry_u_target(void);
+QkTarget *get_rz_ry_u_noerror_target(void);
+bool compare_gate_counts(QkOpCounts counts, char **gates, uint32_t *freq, int num_gates);
+
+/**
+ * Test running pass on chains of h gates.
+ *
+ * Transpile: 0:--[H]-[H]-[H]--
+ */
+int test_optimize_h_gates_inner(QkTarget *target, char **gates, uint32_t *freq, int num_gates) {
+    int result = Ok;
+    // Build circuit
+    QkCircuit *circuit = qk_circuit_new(1, 0);
+    uint32_t qubits[1] = {0};
+    for (int iter = 0; iter < 3; iter++) {
+        qk_circuit_gate(circuit, QkGate_H, qubits, NULL);
+    }
+
+    // Run transpiler pass
+    qk_transpiler_standalone_optimize_1q_gates_decomposition(circuit, target);
+    if (!compare_gate_counts(qk_circuit_count_ops(circuit), gates, freq, num_gates)) {
+        result = EqualityError;
+        goto cleanup;
+    }
+
+cleanup:
+    qk_circuit_free(circuit);
+    qk_target_free(target);
+    return result;
+}
+
+int test_optimize_h_gates(void) {
+    int num_failed = 0;
+    QkTarget *targets[5] = {
+        get_u1_u2_u3_target(), get_rz_rx_target(),           get_rz_sx_target(),
+        get_rz_ry_u_target(),  get_rz_ry_u_noerror_target(),
+    };
+    char *gates[5][2] = {{
+                             "u2",
+                         },
+                         {"rz", "rx"},
+                         {"rz", "sx"},
+                         {"u"},
+                         {"u"}};
+
+    uint32_t freq[5][2] = {
+        {
+            1,
+        },
+        {2, 1},
+        {2, 1},
+        {1},
+        {1},
+    };
+
+    int num_gates[5] = {1, 2, 2, 1, 1};
+    char *names[5] = {"u1_u2_u3", "rz_rx", "rz_sx", "rz_ry_u", "rz_ry_u_noerror"};
+    printf("Optimize h gates tests.\n");
+    for (int idx = 0; idx < 5; idx++) {
+        int result =
+            test_optimize_h_gates_inner(targets[idx], gates[idx], freq[idx], num_gates[idx]);
+        printf("--- Run with %-21s: %s \n", names[idx], (bool)result ? "Fail" : "Ok");
+        num_failed += result;
+    }
+    return num_failed;
+}
+
+int test_optimize_identity_target_inner(QkTarget *target) {
```

Comment:
So far all functions called `test_..` are the main test entry points. I got a little confused when reading this as I expected the same. Could we rename this to something like `check_optimize_identity_with_target(QkTarget* target)` or something to clarify it's a helper?

Reply:
Addressed in 19f55a61197d9d6ac5b188778e4a2dd98b020b1c